contracts/modules/TreasuryModule.sol

```diff
@@ -313,16 +313,11 @@ contract TreasuryModule is
             uint256 netPayoutAmount
         )
     {
-        IPolicy.Payout memory payout =  _policy.getPayout(processId, payoutId);
-        require(
-            payout.state == IPolicy.PayoutState.Expected, 
-            "ERROR:TRS-040:PAYOUT_ALREADY_PROCESSED"
-        );
-
         IPolicy.Metadata memory metadata = _policy.getMetadata(processId);
         IERC20 token = getComponentToken(metadata.productId);
         (uint256 riskpoolId, address riskpoolWalletAddress) = _getRiskpoolWallet(processId);
 
+        IPolicy.Payout memory payout =  _policy.getPayout(processId, payoutId);
```

Comment:
What about the state check `payout.state == IPolicy.PayoutState.Expected`, is this no longer required?

Reply:
the check is already done in PolicyController.processPayout which is called by PolicyDefaultFlow before it calls TreasuryModule.processPayout

the only module allowed to call this treasury function is the PolicyDefaultFlow. inside these functions these checks need to be performed.